feat: Completed navbar, header banner, categories, today best deals

StanleyHaw opened the pull request:
Added:
- Completed the layout of the navigation bar UI, divided into upper and lower sections.
- Completed the header banner.
- Completed the categories section and CSS interactive effects.
- Completed the 'Todays Best Deals For You!' section and JavaScript interactive effects.

Refactored:
- Changed the 'Todays Best Deals For You!' id of carousel-card-button to a class

TODO:
- Navigation bar links are not yet connected; shopping cart functionality is incomplete.
- Header banner links are not set.
- Categories section links are not set.
- 'Todays Best Deals For You!' section API integration pending.

StanleyHaw committed:
feat: Completed navbar, header banner, categories, today best deals

StanleyHaw committed:
refactor: carousel-card-button

AlbertNotAlien commented on script.js:
```diff
@@ -0,0 +1,39 @@
+
+let isDragging = false;
+let startX = null;
+let startScrollLeft = null;
```
startX, startScrollLeft 這兩個如果是數字的話可以用 0 當 default 而不是 null

AlbertNotAlien commented on script.js:
```diff
@@ -0,0 +1,39 @@
+  button.addEventListener('click', () => {
+    if (button.className.includes('card-button-left')) {
+      carousel.scrollLeft -= firstCardWidth + CAROUSEL_GRID_GAP; // TODO: 如何連動scss第272的gap屬性
+    } else {
```
建議可以用 else if (button.className.includes('card-button-right') 會更嚴謹

AlbertNotAlien commented on script.js:
```diff
@@ -0,0 +1,39 @@
+const carousel = document.querySelector('.best-deals-carousel');
+const arrowButtons = document.querySelectorAll('.carousel-card-button i');
+const firstCardWidth = carousel.querySelector('.best-deals-card').offsetWidth;
+const CAROUSEL_GRID_GAP = 16;
```
感覺可以多空一行（？）

AlbertNotAlien commented on script.js:
```diff
@@ -0,0 +1,39 @@
+let startX = null;
+let startScrollLeft = null;
+
+arrowButtons.forEach((button) => {
```
這裡只有分左右的話好像不用特地用 querySelectorAll，分別用 querySelector 感覺會更好理解，這樣也就不用有 button.className.includes 的寫法

StanleyHaw committed:
feat:Complet
ed choose by brand, get up to 70% off, weekly popular products, discount banner

StanleyHaw committed:
feat:Completed footer

StanleyHaw committed:
feat:Completed home page RWD

StanleyHaw committed:
refactor:refactored  item cards carousel javascript
